src/main/java/org/apache/mesos/hdfs/HDFSScheduler.java

```diff
@@ -81,11 +100,11 @@ public void offerRescinded(SchedulerDriver driver, OfferID offerId) {
   public void registered(SchedulerDriver driver, FrameworkID frameworkId, MasterInfo masterInfo) {
     try {
       persistentState.setFrameworkId(frameworkId);
+      log.info("Registered framework frameworkId=" + frameworkId.getValue());
+
     } catch (InterruptedException | ExecutionException e) {
       log.error("Error setting framework id in persistent state", e);
-      throw new RuntimeException(e);
```

Comment:
Why remove the runtime exception? Should you maybe retry on the update? Unsuccessfully persisting your frameworkId means you cannot reregister and reconnect to your tasks if the scheduler suddenly dies. Better to exit before launching any tasks than to potentially lose access to tasks/data.